### PR TITLE
Add styling for shapes on map

### PR DESCRIPTION
**Pull request by Zwatotem**
Adds ability to customize polygons and polylines. Circes are on the way.

### CODE REVIEW

**Comment by Zwatotem** [comment]
I am going to add one bool to these data transfer types (isGeodesic), but other than that this is ready for testing. Please test especially Android environment, as I cannot currently do it.

File: example/App.tsx [hunk 1]
```diff
@@ -1,25 +1,31 @@
-import { StatusBar } from 'expo-status-bar';
-import React from 'react';
-import { StyleSheet, View } from 'react-native';
-import exampleMapStyle from './exampleMapStyle.json';
+import { StatusBar } from "expo-status-bar";
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import exampleMapStyle from "./exampleMapStyle.json";
```
Comment (szdziedzic) [inline example/App.tsx]: We should all use same prettier config, I added some basic config in newest commit, you can rebase to use it so we don't do these unnecessary changes back and forth

File: expo-maps/android/src/main/java/expo/modules/maps/GoogleMapsPolylines.kt [hunk 2]
```diff
@@ -27,4 +33,30 @@ class GoogleMapsPolylines(map: GoogleMap) : Polylines {
...
+    return when (patternItem.type) {
+     PatternItemType.gap -> Gap(patternItem.length)
+     PatternItemType.stroke -> when (patternItem.length) {
+       0F, -0F -> Dot()
+       else -> Dash(patternItem.length)
+     }
+    }
+  }
+
+  private fun jointToNative(joint: Joint) :Int {
```
Comment (szdziedzic) [inline expo-maps/android/src/main/java/expo/modules/maps/GoogleMapsPolylines.kt]:
```suggestion
  private fun jointToNative(joint: Joint): Int {
```

File: expo-maps/android/src/main/java/expo/modules/maps/GoogleMapsPolylines.kt [hunk 3]
```diff
@@ -27,4 +33,30 @@ class GoogleMapsPolylines(map: GoogleMap) : Polylines {
...
+    return when (joint) {
+      Joint.miter -> JointType.DEFAULT
+      Joint.bevel -> JointType.BEVEL
+      Joint.round -> JointType.ROUND
+    }
+  }
+
+  private fun capToNative(cap: Cap) :com.google.android.gms.maps.model.Cap {
```
Comment (szdziedzic) [inline expo-maps/android/src/main/java/expo/modules/maps/GoogleMapsPolylines.kt]:
```suggestion
  private fun capToNative(cap: Cap): com.google.android.gms.maps.model.Cap {
```

File: expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt [hunk 4]
```diff
@@ -25,6 +26,50 @@ data class MarkerObject(
...
+}
 
-data class PolylineObject(@Field val points: Array<Point> = emptyArray()) : Record
+data class CircleObject(
+  @Field val center: Point,
+  @Field val radius: Double,
+  @Field val strokeColor: Int?,
+  @Field val strokeWidth: Float?,
+  @Field val fillColor: Int?,
+) : Record
```
Comment (szdziedzic) [inline expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt:75]:
```suggestion
) : Record

```
